core/services/chainlink/config_test.go

```diff
@@ -1175,7 +1183,7 @@ func TestNewGeneralConfig_Logger(t *testing.T) {
 	}{
 		{name: "empty", wantEffective: emptyEffectiveTOML},
 		{name: "full", inputSecrets: secretsFullTOML, inputConfig: fullTOML,
-			wantConfig: fullTOML, wantEffective: fullTOML, wantSecrets: secretsFullRedactedTOML},
+			wantConfig: fullTOML, wantEffective: fullTOMLEffective, wantSecrets: secretsFullRedactedTOML},
```

Comment:
Is `fullTOML` different than `fullTOMLEffective`? IIRC they were the same by design since no defaults would be applied.

Reply:
we pass fullTOML in the input config and insecure config is not allowed as input on non-dev builds, this is why I separated them. another notion to avoid colliding with `effective` notion is to have fullTOMLInput and fullTOML

Reply:
Hmmm I'm still not sure that I am following. Why can't we choose to test the full config set here? Is validation getting in the way or something?

Reply:
Because our goal at this level is to validate parsing every field, and that they do not get overwritten by a default value when set.

Reply:
I understand, but this [line](https://github.com/smartcontractkit/chainlink/blob/BCF-2060-chainlink-node-offer-insecure-builds/core/services/chainlink/config_general.go#L93) disallows users from setting insecure opts on non-dev builds. Meaning if fullTOML has them, ParseConfig will fail. Alternatively to the current approach, we can emit a warning that the value is going to be ignored (because it will be ignored in [later stages](https://github.com/smartcontractkit/chainlink/blob/BCF-2060-chainlink-node-offer-insecure-builds/core/services/chainlink/config_general.go#L1060) as a safeguard )

I'm indifferent to both options, but leaning toward disallowing it to be on the toml of non-dev builds all-together, this make less room for mistakes (rerunning same toml on dev builds, different behavior)

Reply:
I am in favor of strict validation, but parsing the types/structure of the config and validating the values are separate right now and this sort of tangles them together. Would it make sense to move that check in to the `ValidateConfig()` method?